cmd/stringer/golden_test.go

```diff
@@ -296,11 +327,222 @@ func (i Token) String() string {
 }
 `
 
+// Simple test: enumeration of type int starting at 0.
+const month_in = `type Month int
+const (
+	January Month = iota
+	February
+	March
+	April
+	May
+	June
+	July
+	August
+	September
+	October
+	November
+	December
+)
+`
+
+const month_out = `
+import "strconv"
+
+const _Month_name = "JanuaryFebruaryMarchAprilMayJuneJulyAugustSeptemberOctoberNovemberDecember"
+
+var _Month_index = [...]uint8{0, 7, 15, 20, 25, 28, 32, 36, 42, 51, 58, 66, 74}
+
+func (i Month) String() string {
+	if i < 0 || i >= Month(len(_Month_index)-1) {
+		return "Month(" + strconv.FormatInt(int64(i), 10) + ")"
+	}
+	return _Month_name[_Month_index[i]:_Month_index[i+1]]
+}
+
+func MonthFromString(s string) (i Month, ok bool) {
+	for v := range _Month_index {
+		v += 0
+		if s == Month(v).String() {
+			return Month(v), true
+		}
+	}
+	return
+}
+`
+
+// Enumeration with an negative offset.
+const mood_in = `type Mood int
```

Comment:
warning: don't use underscores in Go names; const mood_in should be moodIn (golint)